src/libs/ws/session.rs

```diff
@@ -168,3 +188,44 @@ impl<
         Ok(())
     }
 }
+
+fn check_roles(role: u32, allowed_roles: &Option<HashSet<u32>>) -> bool {
+    if let Some(allowed_roles) = allowed_roles {
+        if allowed_roles.is_empty() {
```

Comment:
Not required as `contains` will do the same under the hood, so there's no perf gains. Just more complicated code.